internal/fs/vss_windows.go

```diff
@@ -0,0 +1,1061 @@
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	err = callAsyncFunctionAndWait(iVssBackupComponents.GatherWriterMetadata,
+		"GatherWriterMetadata", timeoutInMillis)
+	if err != nil {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	if isSupported, err := iVssBackupComponents.IsVolumeSupported(volume); err != nil {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	} else if !isSupported {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, newVssTextError(fmt.Sprintf("Snapshots are not supported for volume "+
+			"%s", volume))
+	}
+
+	snapshotSetID, err := iVssBackupComponents.StartSnapshotSet()
+	if err != nil {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	if err := iVssBackupComponents.AddToSnapshotSet(volume, &snapshotSetID); err != nil {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	mountPoints, err := enumerateMountedFolders(volume)
+	if err != nil {
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, newVssTextError(fmt.Sprintf(
+			"failed to enumerate mount points for volume %s: %s", volume, err))
+	}
+
+	mountPointInfo := make(map[string]MountPoint)
+
+	for _, mountPoint := range mountPoints {
+		// ensure every mountpoint is available even without a valid
+		// snapshot because we need to consider this when backing up files
+		mountPointInfo[mountPoint] = MountPoint{isSnapshotted: false}
+
+		if isSupported, err := iVssBackupComponents.IsVolumeSupported(mountPoint); err != nil {
+			continue
+		} else if !isSupported {
+			continue
+		}
+
+		var mountPointSnapshotSetID ole.GUID
+		err := iVssBackupComponents.AddToSnapshotSet(mountPoint, &mountPointSnapshotSetID)
+		if err != nil {
+			iVssBackupComponents.Release()
+			return VssSnapshot{}, err
+		}
+
+		mountPointInfo[mountPoint] = MountPoint{isSnapshotted: true,
+			snapshotSetID: mountPointSnapshotSetID}
+	}
+
+	err = callAsyncFunctionAndWait(iVssBackupComponents.PrepareForBackup, "PrepareForBackup",
+		timeoutInMillis)
+	if err != nil {
+		// After calling PrepareForBackup one needs to call AbortBackup() before releasing the VSS
+		// instance for proper cleanup.
+		// It is not neccessary to call BackupComplete before releasing the VSS instance afterwards.
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	err = callAsyncFunctionAndWait(iVssBackupComponents.DoSnapshotSet, "DoSnapshotSet",
+		timeoutInMillis)
+	if err != nil {
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	var snapshotProperties VssSnapshotProperties
+	err = iVssBackupComponents.GetSnapshotProperties(snapshotSetID, &snapshotProperties)
+	if err != nil {
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	for mountPoint, info := range mountPointInfo {
```

Comment:
I think you need check for `info.isSnapshotted` here, otherwise you report meaningless (you call `GetSnapshotProperties` with empty `snapshotSetID`) errors for mountpoints skipped in lines 839-843.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/04c149123db9765258feddadd5a46a93484f66af